github-import-wiki/python-runnables/github-wiki/runnable.py

```diff
@@ -0,0 +1,125 @@
+        self.project_key = project_key
+        self.config = config
+        self.plugin_config = plugin_config
+        
+    def get_progress_target(self):
+        """
+        If the runnable will return some progress info, have this function return a tuple of 
+        (target, unit) where unit is one of: SIZE, FILES, RECORDS, NONE
+        """
+        return (100, None)
+
+    def run(self, progress_callback):
+        """
+        Do stuff here. Can return a string or raise an exception.
+        The progress_callback is a function expecting 1 value: current progress
+        """
+        rt = ResultTable()
+        rt.add_column("article", "Article", "STRING")
+
+        client = dataiku.api_client()
+        wiki = DSSWiki(client, self.project_key)
+        
+        print 'self.config'
+        print self.config
+
+        repository_url = 'https://github.com/dataiku/dip'
+        if 'repository_url' in self.config:
+            repository_url = self.config['repository_url']
+            
+        wiki_url = repository_url + '/wiki'
+        clone_url = repository_url + '.wiki.git'
+        if 'username' in self.config and self.config['username'] != '' and 'token' in self.config and self.config['token'] != '':
+            username = self.config['username']
+            token = self.config['token']
+            wiki_url_parts = repository_url.split('://')
+            wiki_url_parts.insert(1, username + ":" + token + '@')
+            wiki_url_parts.insert(1, '://')
+            clone_url = ''.join(wiki_url_parts) + '.wiki.git'
+                    
+        directory = tempfile.mkdtemp()
+        
+        if os.path.isdir(directory):
+            shutil.rmtree(directory)
+        cloneResult = subprocess.call(["git", "clone", clone_url, directory])
+        if cloneResult != 0:
+            raise Exception('Unable to clone repository. Check your username and token or password.')
+                
+        files = []
+        urls = []
+        for (dirpath, dirnames, filenames) in walk(directory):
+            for filename in filenames:
+                if filename.endswith('.md'):
+                    files.append(filename[:-3])
+            break
+
+        articles = wiki.list_articles()
+        article_ids = []
+        for article in articles:
+            article_ids.append(article.article_id)
+
+        for filename in files:
+            file = open(directory + '/' + filename + '.md', "r")
+            
+        total_files = len(files)
+        processed_files = 0
+        for filename in files:
+            file = codecs.open(directory + '/' + filename + '.md', encoding='utf-8')
+            print 'Importing ' + filename
+            rt.add_record([filename])
+            processed_files = processed_files + 1
+            progress = processed_files*100 / total_files
+            progress_callback(progress)
+
+            content = file.read()
+            
+            pattern = r'(\[.+\]\(' + wiki_url + r'/.+\))'
```

Comment:
This may need an explanation that this is to rewrite absolute links that may exist (I guess ?)